packages/grafana-ui/src/components/Table/TableNG/TableNG.tsx

```diff
@@ -457,14 +460,20 @@ export function TableNG(props: TableNGProps) {
             <Cell
               key={key}
               {...props}
-              className={clsx(props.className, defaultCellStyles, cellSpecificStyles, linkStyles)}
+              className={clsx(props.className, { [cellParentStyles]: maxHeight == null })}
               style={style}
             />
           );
         };
 
         result.cellRootRenderers[displayName] = renderCellRoot;
 
+        const clampByMaxHeight = (maxHeight: number, children: ReactNode, cellStyles: string) => (
+          <div className={clsx(styles.cellClamp, cellStyles)} style={{ maxHeight }}>
```

Comment:
when a max height is present for a cell, we introduce a wrapping div around the cell content because that content may be a single element or it may be a list of child elements, so we can't just apply a class to the some element at that level. Applying a class at `Cell` also doesn't work for this, it breaks the grid to resize the height of that element, so we need an element within the `Cell` element which has a height limit and has the `line-clamp` property on it.